### PR TITLE
refactor(behavior_velocity): refactor occlusion spot module

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Related Issue(required)

#258 

## Description(required)

refactor following (no behavior change except module selection by param)
- remove expression of private road
- remove filter by road type and add detection type param

create instance if path point size > 0
remove instance if path point == 0

This PR wont refactor
- getCurrentRoadType: detailed process of considering lanelet with highway(should the highway occlusion be filtered or we should not making instance at highway or what?) .

## Review Procedure(required)

[launch psim](https://tier4.atlassian.net/wiki/spaces/T4DC/pages/2298019949/Psim)

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guidelines]
- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: planning/behavior_velocity_planner/config/occlusion_spot.param.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /**:
   ros__parameters:
     occlusion_spot:
+      method: "occupancy_grid"             # [-] candidate is "occupancy_grid" or "predicted_object"
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/config/occlusion_spot.param.yaml:4]: memo
default: predicted object

File: planning/behavior_velocity_planner/src/scene_module/occlusion_spot/manager.cpp [hunk 2]
```diff
@@ -117,19 +82,20 @@ OcclusionSpotModuleManager::OcclusionSpotModuleManager(rclcpp::Node & node)
...
+  const int64_t module_id = static_cast<int64_t>(ModuleID::DEFAULT);
   const int64_t public_road_module_id = static_cast<int64_t>(ModuleID::PUBLIC);
-  // private
-  if (!isModuleRegistered(private_road_module_id)) {
-    if (hasPrivateRoadOnPath(path, planner_data_->lanelet_map)) {
-      registerModule(std::make_shared<OcclusionSpotInPrivateModule>(
-        private_road_module_id, planner_data_, planner_param_,
-        logger_.get_child("occlusion_spot_in_private_module"), clock_, pub_debug_occupancy_grid_));
+  // general
+  if (!isModuleRegistered(module_id)) {
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/scene_module/occlusion_spot/manager.cpp:89]: Memo 
way to create instance

File: planning/behavior_velocity_planner/include/scene_module/occlusion_spot/manager.hpp [hunk 3]
```diff
@@ -45,7 +45,7 @@ class OcclusionSpotModuleManager : public SceneModuleManagerInterface
   const char * getModuleName() override { return "occlusion_spot"; }
 
 private:
-  enum class ModuleID { PRIVATE, PUBLIC };
+  enum class ModuleID { DEFAULT, PUBLIC };
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/include/scene_module/occlusion_spot/manager.hpp:48]: Memo
default or general ?

Comment (taikitanaka3) [reply planning/behavior_velocity_planner/include/scene_module/occlusion_spot/manager.hpp]: I change to GENERAL [cb73c4f](https://github.com/autowarefoundation/autoware.universe/pull/400/commits/cb73c4fce060c7a462180b6ed630a02fe41c8462)

File: planning/behavior_velocity_planner/include/scene_module/occlusion_spot/manager.hpp [hunk 4]
```diff
@@ -45,7 +45,7 @@ class OcclusionSpotModuleManager : public SceneModuleManagerInterface
   const char * getModuleName() override { return "occlusion_spot"; }
 
 private:
-  enum class ModuleID { PRIVATE, PUBLIC };
+  enum class ModuleID { GENERAL, PUBLIC };
```
Comment (tkimura4) [inline planning/behavior_velocity_planner/include/scene_module/occlusion_spot/manager.hpp:48]: I think ModuleID is better to be simply OCCUPANCY/OBJECT corresponding with method names.

**Comment by tkimura4** [comment]
Could you update readme too?

**Comment by tkimura4** [comment]
you can remove `extractTargetRoadArcLength` because this function  is no longer used.

File: planning/behavior_velocity_planner/src/scene_module/occlusion_spot/manager.cpp [hunk 5]
```diff
@@ -141,17 +107,14 @@ std::function<bool(const std::shared_ptr<SceneModuleInterface> &)>
...
-    }
-    if (scene_module->getModuleId() == static_cast<int64_t>(ModuleID::PUBLIC)) {
-      return !has_public_road;
-    }
-    return true;
+  return [path]([[maybe_unused]] const std::shared_ptr<SceneModuleInterface> & scene_module) {
+    // expire if no path points
+    if (path.points.empty()) return true;
+    // expire if there are path points
+    else if (path.points.size() > 0)
```
[the remark this answers is not in the document]
Comment (taikitanaka3) [reply planning/behavior_velocity_planner/src/scene_module/occlusion_spot/manager.cpp]: maybe unregister won't be necessary [000cae9](https://github.com/autowarefoundation/autoware.universe/pull/400/commits/000cae9f68725f67d23cbaa7027a0d2d36f2b954)

File: planning/behavior_velocity_planner/include/scene_module/occlusion_spot/manager.hpp (hunk 4 above)
Comment (taikitanaka3) [reply planning/behavior_velocity_planner/include/scene_module/occlusion_spot/manager.hpp]: changed to occupancy and object
[000cae9](https://github.com/autowarefoundation/autoware.universe/pull/400/commits/000cae9f68725f67d23cbaa7027a0d2d36f2b954)

**Comment by taikitanaka3** [comment]
rebased for conflict with interpolation refactoring

**Comment by tkimura4** [comment]
method: predicted object
image

method: occupancy grid
image

**Review by tkimura4 — APPROVED** [review]
LGTM

**Comment by taikitanaka3** [comment]
@yukkysaito 
can you approve this if this is ok?

**Review by yukkysaito — APPROVED** [review]

File: sensing/laserscan_to_occupancy_grid_map/launch/laserscan_to_occupancy_grid_map.launch.py [hunk 6]
```diff
@@ -61,7 +61,7 @@ def add_launch_arg(name: str, default_value=None):
                     "max_height": 2.0,
                     "angle_min": -3.141592,  # -M_PI
                     "angle_max": 3.141592,  # M_PI
-                    "angle_increment": 0.00436332222,  # 0.25*M_PI/180.0
+                    "angle_increment": 0.00349065850,  # 0.20*M_PI/180.0
```
Comment (yukkysaito) [inline sensing/laserscan_to_occupancy_grid_map/launch/laserscan_to_occupancy_grid_map.launch.py:64]: I've been using this in dummy perception launch, and I want to make it common with perception launch.

Comment (taikitanaka3) [reply sensing/laserscan_to_occupancy_grid_map/launch/laserscan_to_occupancy_grid_map.launch.py:64]: change for dummy_perception_publisher_launch to refer perception launch will be at  another PR

**Comment by taikitanaka3** [comment]
- [x] release note

**Pull request merged**